### ISSUE TITLE
Shapes with 3 vertices in webgl mode are always closed

### ISSUE THREAD

**Issue by franolichdesign** [processing/p5.js#7379]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [X] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [ ] Friendly errors
- [ ] Other (specify if possible)

### p5.js version

1.11.1

### Web browser and version

Firefox 132.0.2 / Edge 130.0.2849.80

### Operating system

Windows 11 Home / Pro

### Steps:
1.Run the following code [(p5 editor)](https://editor.p5js.org/franolich/sketches/2ygDbsyGQ). The triangle at the top-left of the canvas should be an L-shaped corner but has been closed even though `endShape()` is called without the `CLOSE` constant.
2.To see the desired result, change the value of the `webgl` variable in line 1 to false and run the code again.

### Snippet:

```js

let webgl = true;

function setup() {
  if (webgl) {
    createCanvas(400, 400, WEBGL);
  } else {
    createCanvas(400, 400);
  }
  
  background(220);  
  stroke(0);
  strokeWeight(5);
  noFill();

  let offset = (webgl ? -100 : 100);
  translate(offset, offset);
  beginShape();
  vertex(100, 0, 0);
  vertex(0, 0, 0);
  vertex(0, 100, 0);
  endShape();
  
  translate(100, 100);
  beginShape();
  vertex(100, 0, 0);
  vertex(0, 0, 0);
  vertex(0, 100, 0);
  vertex(100, 100, 0);
  endShape();
}

```

**Comment by rishabhs-rizz** [comment]
hey, i would like to work on this issue. As I am a newcomer in the open source, i will try to  get a boost in my confidence by solving this issue.

**Comment by franolichdesign** [comment]
Hi! Thanks for taking a look at this. Unfortunately I do not have
permission to assign the issue on GitHub. As a p5/GitHub noobie I'm not
sure how to proceed so I'll follow your suggestions.
Cheers,
Fran

**Comment by franolichdesign** [comment]
@rishabhs-rizz - The help page [Assigning issues and pull requests to other GitHub users](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/assigning-issues-and-pull-requests-to-other-github-users) states that "Anyone with write access to a repository can assign issues and pull requests." ie not me! I guess we'll just have to wait for for a project maintainer to look at the issue.

**Comment by davepagurek** [comment]
Thanks for noticing this issue! So the problematic code is also a piece that is being currently refactored for the upcoming 2.0 release, the code for which is on the `dev-2.0` branch. It's a little harder to contribute to that, but if you're up to it, you could make a fork off of that branch and make a PR into that branch.

The offending code, I think, is this bit here, that converts to TRIANGLES mode if there are just 3 points for the sake of speed: https://github.com/processing/p5.js/blob/be504ea18311d85950e6de0929a8d0481bf7ddfa/src/webgl/ShapeBuilder.js#L69-L76

We maybe just need to add an extra condition to this if statement to only do this optimization if `endShape` has been called with `CLOSE`.

**Comment by davepagurek** [comment]
Awesome, thanks! I'll assign this to you.

**Comment by rishabhs-rizz** [comment]
@davepagurek sir, i am facing an issue in dev-2.0 branch. when i am running "npm run dev:global" command  this is getting showed

![p5](https://github.com/user-attachments/assets/a7c93cba-ff0a-4806-b0d6-f8d888601354)


how can I see the changes i made,
because i think that putting an extra if-condition will resolve the issue.

**Comment by davepagurek** [comment]
If you edit the sketch in `preview/global/sketch.js` or modify anything in the `src/` directory, while the dev server is running, you should see the sketch auto-reload in the browser and run again with your new changes.

**Comment by davepagurek** [comment]
I'm going to close this for now, after our shapes refactor, that particular optimization is no longer present in the codebase. Let us know if you notice any new issues in the 2.0 branch though!

**Comment by rishabhs-rizz** [comment]
> I'm going to close this for now, after our shapes refactor, that particular optimization is no longer present in the codebase. Let us know if you notice any new issues in the 2.0 branch though!

You're right, when I tried to fix this in dev-2.0 branch. It was already working fine there.
